src/backups.py

```diff
@@ -151,12 +151,21 @@ def _on_create_backup_action(self, event) -> None:
 
         try:
             self.charm.unit.status = MaintenanceStatus("creating backup")
+
+            # Remove the pod labels to hide this unit from the replicas K8S service.
+            if self.charm.app.planned_units() > 1:
+                self.charm.patch_pod_labels(self.charm.unit.name, add_labels=False)
```

Comment:
maybe additionally we need to run `update pg_database set datallowconn = false where datname = 'databasename';` to hide db instance from pgbouncer

Reply:
After testing it, I saw that it only takes effect after connections are closed and new connections are created on PgBouncer side. So it doesn’t block new connections from the client applications immediately. Also, it blocks connections on all instances because the database is the same for reads and writes on all instances.